app/src/org/commcare/views/media/MediaLayout.java

```diff
@@ -114,69 +104,57 @@ public static MediaLayout buildComprehensiveLayout(Context context,
+//                                                       int questionIndex) {
+//        setAVT(text, audioURI, imageURI, videoURI, bigImageURI, qrCodeContent, inlineVideoURI, false, questionIndex);
+//    }
+
+
+
+    public void addDivider() {
+        divider.setVisibility(VISIBLE);
+    }
+
+    //region private helpers
+
     private void setAVT(TextView text, String audioURI, String imageURI,
                         final String videoURI, final String bigImageURI,
                         final String qrCodeContent, String inlineVideoURI,
                         boolean showImageAboveText,
                         int questionIndex) {
-        viewText = text;
-        mInlineVideoUri = inlineVideoURI;
-        mImageURI = imageURI;
-        mBigImageURI = bigImageURI;
-        mQrCodeContent = qrCodeContent;
-
-        RelativeLayout questionTextPane = new RelativeLayout(this.getContext());
-        questionTextPane.setId(QUESTION_TEXT_PANE_ID);
-
         setupStandardAudio(audioURI, questionIndex);
         setupVideoButton(videoURI);
-
-        // Now set up the center view -- it is either an image, a QR Code, an inline video, or
-        // expanded audio
-        mediaPane = new RelativeLayout(getContext());
-
-        LayoutParams mediaPaneParams = refreshMediaView();
-
-        addAudioVideoButtonsToView(questionTextPane);
-
-        showImageAboveText = showImageAboveText || DeveloperPreferences.imageAboveTextEnabled();
-        addElementsToView(mediaPane, mediaPaneParams, questionTextPane, showImageAboveText);
+        setupQRView(qrCodeContent);
+        setupInlineVideoView(inlineVideoURI);
+        setupImage(imageURI, bigImageURI);
+        addTextView(text);
+        //TODO Need to use showImageAboveText
     }
 
-    private LayoutParams refreshMediaView() {
-        RelativeLayout.LayoutParams mediaPaneParams =
-                new RelativeLayout.LayoutParams(LayoutParams.MATCH_PARENT, LayoutParams.WRAP_CONTENT);
-
-        View mediaView = null;
-        if (mInlineVideoUri != null) {
-            mediaView = getInlineVideoView(mInlineVideoUri, mediaPaneParams);
-        } else if (mQrCodeContent != null) {
-            mediaView = setupQRView(mQrCodeContent);
-        } else if (mImageURI != null) {
-            mediaView = setupImage(mImageURI, mBigImageURI);
-        }
+    private void addTextView(TextView text) {
+        textViewContainer.addView(text);
+    }
 
-        if (mediaView != null) {
-            RelativeLayout.LayoutParams mediaViewParams =
-                    new RelativeLayout.LayoutParams(LayoutParams.WRAP_CONTENT, LayoutParams.WRAP_CONTENT);
-            mediaViewParams.addRule(CENTER_IN_PARENT, mediaView.getId());
-            mediaPane.addView(mediaView, mediaViewParams);
+    private void setupStandardAudio(String audioURI, int questionIndex) {
+        if (audioURI != null) {
+            boolean mediaPresent = FileUtil.referenceFileExists(audioURI);
```

Comment:
Wondering if we need to do something here?

Reply:
the audio button should be handling the missing media state by itself, providing an option to download the file in case it's missing.

Reply:
awesome. Thanks!

Reply:
FYI, I'll remove this unused variable then.